File: module-load-balance-gateway/src/main/java/com/xmmufan/permission/service/UserInfoService.java
package com.xmmufan.permission.service;


import com.xmmufan.permission.domain.rbac.User;
import com.xmmufan.permission.domain.rbac.UserInfo;

public interface UserInfoService {

    void saveOrUpdate(UserInfo userInfo);

    UserInfo queryByUser(User user);

    UserInfo queryById(String id);

}

File: module-load-balance-gateway/src/main/java/com/xmmufan/permission/service/UserInfoService.kt
package com.xmmufan.permission.service


import com.xmmufan.permission.domain.rbac.UserInfo

interface UserInfoService {

    fun saveOrUpdate(userInfo: UserInfo, userId: String)

    fun queryByUser(userId: String): UserInfo?

    fun queryById(id: String): UserInfo?

}
